### ISSUE TITLE
post logout redirect session endpoint 

### ISSUE THREAD

**Issue by Johnpc123** [panva/node-openid-client#70]
I'm using this client with https://github.com/panva/node-oidc-provider.  

After some success implementing oidc flows for authentication I turned off the experimental sessionManagment feature because it started throwing an error.   

In the get.logout route on the client I changed the following to work around the error.  

```
      ctx.redirect(url.format(Object.assign(url.parse(issuer.end_session_endpoint), {
        search: null,
        query: {
          id_token_hint: tokens.id_token,
          post_logout_redirect_uri: url.resolve(ctx.href, '/'),
        },
      }))); 
```
to 

```
   if (issuer.end_session_endpoint) {
      ctx.redirect(url.format(Object.assign(url.parse(issuer.end_session_endpoint), {
        search: null,
        query: {
          id_token_hint: tokens.id_token,
          post_logout_redirect_uri: url.resolve(ctx.href, '/'),
        },
      })));
    } else ctx.redirect('/');
```
I'll be delighted when I understand the session management feature well enough to stop getting the error.  the last time I saw it it was because I'd inadvertently introduced a mismatch in urls between the OP and RP.  Currently it's more expedient to simply turn off the feature and leave session management alone for the moment.     

Maybe it's trivial, but I was looking for any way to say thank you for making the oidc provider and client available.  I thought you might consider this edit as helps the 'default' behaviour of the client.    

- thank you, it's a wonderful package!

**Comment by panva** [comment]
Hi @Johnpc123, thank you for the kind words, i'm glad the libs see more and more use. 

I'd be interested in the errors you're seeing in the oidc-provider issues section. As far as the client goes - the example is written with the feature enabled in mind and so obviously (when sessionManagement gets disabled) the features have to too. The example is merely intended to showcase the API in use.

**Comment by Johnpc123** [comment]
Thanks for the prompt reply.  
Currently I simply see an alert box popup in the User Agent after logging on.   I'm guessing that if I turn on the debugging feature on the OP I may get some specific details. 

When I get a chance I'll try that, check out the error and update.

**Comment by panva** [comment]
Hey @Johnpc123, what you're seeing is an implementation which, granted when you don't know the session management spec, features, and how to use them, is useless to you. I sincerely hope you didn't use the example as a base for your project. It is simply showing the public API in use but is NOT a complete project and does not come with essentials such as proper sessions, basic web application protection etc.

Bottom line, the alert is not an error, but merely a showcase of the session management [Session Status Change Notification](https://openid.net/specs/openid-connect-session-1_0.html#ChangeNotification) working.

**Comment by Johnpc123** [comment]
Thanks for the warnings Philip, I'll be sure to balance risks objectively.
